server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -349,18 +349,24 @@ public Query rangeQuery(
             }
 
             @Override
-            public List<Field> createFields(String name, Number value, boolean indexed, boolean docValued, boolean stored) {
-                List<Field> fields = new ArrayList<>();
+            public void createFields(
+                Consumer<IndexableField> consumer,
```

Comment:
I wonder if it is better to supply list at first place to avoid creating consumer from `List#add()`?

Reply:
Refactored to using `LuceneDocument` directly here without any indirection now :)